js/core/src/init.svelte.ts

```diff
@@ -77,16 +77,18 @@ export class AppTree {
 		components: ComponentMeta[],
 		layout: LayoutNode,
 		dependencies: Dependency[],
-		config: AppConfig,
+		config: Omit<AppConfig, "api_url">,
 		app: client_return,
 		reactive_formatter: (str: string) => string
 	) {
 		this.ready = new Promise<void>((resolve) => {
 			this.ready_resolve = resolve;
 		});
 		this.reactive_formatter = reactive_formatter;
-
-		this.#config = config;
+		this.#config = {
+			...config,
+			api_url: new URL(config.api_prefix, config.root).toString()
+		};
```

Comment:
Added a new 'url' to the local config which is just the root + api_prefix for use when fetching custom components.